File: server/adapter/api/dev/code.go
package dev

import (
	"fmt"
	"github.com/gin-gonic/gin"
	"os"
	"server/client/common/request"
	"server/client/common/response"
	"server/client/dev/cmd"
)

type CodeAdapter struct {
}

// Page Table
// @Tags     工具模块
// @Summary  查询数据库表
// @Produce   application/json
// @Param    data  query     request.PageQuery     true  "分页参数"
// @Success  200   {object}  response.Response
// @Router   /dev/code/page [get]
func (p *CodeAdapter) Page(c *gin.Context) {
	//绑定参数
	var page request.PageQuery
	err := c.ShouldBindQuery(&page)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	pageResult, err := codeServiceApp.Page(page)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.OkWithData(pageResult, c)
}

// Create Table
// @Tags     工具模块
// @Summary  生成代码
// @Produce   application/json
// @Param    data  body      cmd.CodeCreateCmd     true  "用户信息"
// @Success  200   {object}  response.Response
// @Router   /dev/code/create [post]
func (p *CodeAdapter) Create(c *gin.Context) {
	//绑定参数
	var command cmd.CodeCreateCmd
	err := c.ShouldBindJSON(&command)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	//内部创建
	_, err = codeServiceApp.CreateInner(command)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.Ok(c)
}

// Download Table
// @Tags     工具模块
// @Summary  生成代码
// @Produce   application/json
// @Param    data  body      cmd.CodeCreateCmd 		true 	"生成代码"
// @Success  200   {object}  response.Response
// @Router   /dev/code/download [post]
func (p *CodeAdapter) Download(c *gin.Context) {
	//绑定参数
	var command cmd.CodeCreateCmd
	err := c.ShouldBindJSON(&command)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	//下载
	_, err = codeServiceApp.CreateDownload(command)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
	} else {
		c.Writer.Header().Add("Content-Disposition", fmt.Sprintf("attachment; filename=%s", "code.zip"))
		c.File("./code.zip")
		os.Remove("./code.zip")
	}
}

// Delete Table
// @Tags     工具模块
// @Summary  删除最近一次生成的代码
// @Produce   application/json
// @Success  200   {object}  response.Response
// @Router   /dev/code/delete [delete]
func (p *CodeAdapter) Delete(c *gin.Context) {
	err := codeServiceApp.Delete()
	if err != nil {
		response.FailWithMessage(err.Error(), c)
	} else {
		response.Ok(c)
	}
}

File: web/src/api/dev/code.js
import request from '@/utils/request'

// 分页查询列表
export function pageTable(data) {
  return request({
    url: '/dev/code/page',
    method: 'get',
    params: data
  })
}
// 查询列表
export function listTable(data) {
  return request({
    url: '/dev/code/list',
    method: 'get',
    params: data
  })
}
// 查询User详细
export function getTable(id) {
  return request({
    url: '/dev/code/info',
    method: 'get',
    params: { 'id': id }
  })
}
// 生成代码
export function create(data) {
  return request({
    url: '/dev/code/create',
    method: 'post',
    data: data
  })
}
// DeleteRecent
export function DeleteRecent() {
  return request({
    url: '/dev/code/delete',
    method: 'delete'
  })
}
// 生成代码
export function download(data) {
  return request({
    url: '/dev/code/download',
    method: 'post',
    data: data,
    responseType: 'blob'
  })
}
